src/corva/handlers.py

```diff
@@ -0,0 +1,154 @@
+import contextlib
+import functools
+from typing import Any, Callable, List, Type
+
+from corva.api import Api
+from corva.configuration import SETTINGS
+from corva.logger import setup_logging
+from corva.models.base import RawBaseEvent
+from corva.models.context import CorvaContext
+from corva.models.scheduled import RawScheduledEvent, ScheduledEvent
+from corva.models.stream.raw import RawStreamEvent
+from corva.models.stream.stream import StreamEvent
+from corva.models.task import RawTaskEvent, TaskEvent, TaskStatus
+from corva.state.redis_state import RedisState, get_cache
+
+
+def base_handler(raw_event_type: Type[RawBaseEvent]) -> Callable:
+    def decorator(func: Callable[[RawBaseEvent, Api, str], Any]) -> Callable:
+        @functools.wraps(func)
+        def wrapper(aws_event: Any, aws_context: Any) -> List[Any]:
+            context = CorvaContext.parse_obj(aws_context)
+
+            api = Api(
+                api_url=SETTINGS.API_ROOT_URL,
+                data_api_url=SETTINGS.DATA_API_ROOT_URL,
+                api_key=context.api_key,
+                app_name=SETTINGS.APP_NAME,
```

Comment:
Not related to this ticket but `X-Corva-App` should be `APP_KEY` instead of  `APP_NAME`

Reply:
Fixed, this was a ticket. [DC-1046](https://corvaqa.atlassian.net/browse/DC-1046)